src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -816,7 +816,11 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                     pattern = rList[0];
                     if (rList.Count > 1)
                     {
-                        if(rList[1] is ScriptBlock)
+                        if (rList[1] is MatchEvaluator)
```

Comment:
Maybe have a fast path for the string replacement (likely the most common case), and then, if it isn't a string check for
scriptblock, MatchEvaluator, and PSMethod?
If it is a PSMethod, try to convert it to a MatchEvaluator.

Reply:
How about a single branch like:

    if(!(rList[!]] is string))
        Attempt conversion to MatchEvaluator